tests/listeners/test_listener.py

```diff
@@ -32,13 +35,23 @@ class TestListener:
             metrics=metrics,
             condition=condition,
             execution_interval_in_seconds=1,
+            **(query_params or {}),
         )
         def action(self, records: List[Record], ctx: RBListenerContext):
             try:
+                assert ctx.dataset == dataset
+                if ctx.query_params:
+                    assert ctx.query == query.format(**ctx.query_params)
+                    if self.executed:
+                        assert ctx.query_params != query_params
+                    else:
+                        assert ctx.query_params == query_params
+                    ctx.query_params["params"] += 1
```

Comment:
Maybe you can just check the query after changing the query params:

```suggestion
                    ctx.query_params["params"] += 1
                    assert ctx.query_params != query_params
                    assert ctx.query == query.format(**ctx.query_params)
```

Reply:
hm, here the idea was to test if changes to the `ctx.query_params` persist to the next execution of the action. I think i will simply remove the else clause.

Reply:
👍